### PR TITLE
CONTRAST-20253

### PR DESCRIPTION
**Pull request by rduran0**
* If the /api or /Contrast/api suffix is missing at the end of the team
server url, its now added after saving.
* Renamed TeamServer label to Contrast URL.
* Removed warnings from settings. A tooltip was added to Url, service and
API settings.
* Refactorized views creation code for Settings. Reduced many lines in
favor of using UIElementUtils methods for its creation.
* Replaced error label for Organizations dialog with a Message dialog. Now the errors should be displayed and the combo box disabled and cleared.

### CODE REVIEW

File: plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/ContrastPreferencesPage.java [hunk 1]
```diff
@@ -52,13 +52,17 @@
...
 import com.contrastsecurity.ide.eclipse.ui.ContrastUIActivator;
+import com.contrastsecurity.ide.eclipse.ui.util.UIElementUtils;
 import com.contrastsecurity.models.Organization;
 import com.contrastsecurity.sdk.ContrastSDK;
 
 public class ContrastPreferencesPage extends PreferencePage implements IWorkbenchPreferencePage {
 
 	public static final String ID = "com.contrastsecurity.ide.eclipse.ui.internal.preferences.ContrastPreferencesPage";
 	private static final String BLANK = "";
+	private final static String URL_CONTRAST_SUBFIX = "/Contrast";
```
Comment (gmoore) [inline plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/ContrastPreferencesPage.java]: SUBFIX -> SUFFIX

File: plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/OrganizationPreferencesDialog.java [hunk 2]
```diff
@@ -1,318 +1,313 @@
-package com.contrastsecurity.ide.eclipse.ui.internal.preferences;
```
Comment (gmoore) [inline plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/OrganizationPreferencesDialog.java]: Did you mean to replace this whole file? Is it the indentation?

Comment (rduran0) [reply plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/OrganizationPreferencesDialog.java]: Indentation problem

File: plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/ContrastPreferencesPage.java (hunk 1 above)
Comment (rduran0) [reply plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/preferences/ContrastPreferencesPage.java]: Done.

**Comment by serge17** [comment]
The test connection and adding/editing organizations should work if the "/api" is not added to the url.

**Comment by rduran0** [comment]
@serge17 Change done. Now the URL should be completed when you test connection or add/edit an organization.

**Comment by rduran0** [comment]
@gmoore I added one more check for those cases...It should work even if they have an url like this:

https://app.contrastsecurity.com/Contrast/api////

It should remove al the slashes.

**Pull request merged**